Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs

```diff
@@ -386,19 +386,72 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
             foreach (ElementInfo unidentifiedElement in unidentifiedElements)
             {
                 if (unidentifiedElement.ElementStatus == ElementInfo.eElementStatus.Failed)
-                {  
-                    //Deleted
-                    DeltaViewElements.Add(ConvertElementToDelta(unidentifiedElement, eDeltaStatus.Deleted, unidentifiedElement.ElementGroup, true, "Element not found on page"));
+                {
+                    bool matchingElementFound = false;
+
+                    DeltaElementInfo toRemoveAddedFoundItem = null;
+                    foreach (var newElement in addedElements)
+                    {
+                        var newElementLocdiff = newElement.Locators.ToDictionary(x => x.LocateBy, x => x.LocateValue);
+                        var oldElementLocdiff = unidentifiedElement.Locators.ToDictionary(x => x.LocateBy, x => x.LocateValue);
+
+                        if (newElementLocdiff.Except(oldElementLocdiff).Count() == 0)
+                        {
+                            //check property and update
+                            var newElementPropeties = newElement.Properties.ToDictionary(x => x.Name, x => x.Value);
+                            var oldElementPropeties = unidentifiedElement.Properties.ToDictionary(x => x.Name, x => x.Value);
+                            var diffProperties = newElementPropeties.Except(oldElementPropeties);
+
+                            foreach (var changedProprty in diffProperties.ToList())
+                            {
+                                foreach( var existingProperty in unidentifiedElement.Properties.Where(x => x.Name == changedProprty.Key))
+                                {
+                                    existingProperty.Value = changedProprty.Value;
+                                }
+                            }
+
+                            matchingElementFound = true;
+                            DeltaViewElements.Add(ConvertElementToDelta(unidentifiedElement, eDeltaStatus.Changed, unidentifiedElement.ElementGroup, true, "Element Property Updated"));
+                            
+                            var itemToRemove = DeltaViewElements.Where(x => x.ElementInfo.Guid == newElement.ElementInfo.Guid).FirstOrDefault();
+                            if (itemToRemove != null)
+                            {
+                                DeltaViewElements.Remove(itemToRemove);
+                            }
+
+                            // add found newElment in removeitem 
+                            toRemoveAddedFoundItem = newElement;
+                            //element found and updated, so exit from loop
+                            break;
+                        }
+
+                    }
+
+                    if (toRemoveAddedFoundItem != null)
+                    {
+                        addedElements.Remove(toRemoveAddedFoundItem);
+                    }
+
+                    if (matchingElementFound==false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4592337)